airflow/kubernetes/refresh_config.py

```diff
@@ -24,15 +24,22 @@
 import logging
 import os
 import time
-from datetime import datetime
 from typing import Optional
 
+import pendulum
 import yaml
 from kubernetes.client import Configuration
 from kubernetes.config.exec_provider import ExecProvider
 from kubernetes.config.kube_config import KUBE_CONFIG_DEFAULT_LOCATION, KubeConfigLoader
 
 
+def _parse_timestamp(ts_str: str) -> int:
+    parsed_dt = pendulum.parse(ts_str)
+    if isinstance(parsed_dt, pendulum.DateTime):
```

Comment:
What's this for? Doesn't pendulum throw an error if it can't parse?

Reply:
The problem is, that pendulum.parse type annotations says it return `typing.Union[Date, Time, DateTime, Duration]`, however not all of these implement `timetuple()`.
One of the CI tests (I guess it was mypy) yelled at me for not covering the other possible return types. I was not able to find an input string which didn't result in either a parser error or in a return type DateTime so it doesn't really matter I guess.

Reply:
Ah, `cast` from the typing module or a `# type:` comment is probably the way to go then

Reply:
Ah, yes, cast sounds like it was made exactly for cases like this. Never heard about it before. I adjusted the code.